lib/cancan/conditions_matcher.rb

```diff
@@ -78,7 +78,7 @@ def condition_match?(attribute, value)
 
     def hash_condition_match?(attribute, value)
       if attribute.is_a?(Array) || (defined?(ActiveRecord) && attribute.is_a?(ActiveRecord::Relation))
-        attribute.any? { |element| matches_conditions_hash?(element, value) }
+        attribute.to_a.any? { |element| matches_conditions_hash?(element, value) }
```

Comment:
@Liberatys if I remove the `.to_a` here, all tests still pass locally for me.

Do you know if I am missing something? 

I've tried testing against AR 5.2 and 6.0 on sqlite and postgres.

Reply:
@ghiculescu the tests are mostly written to ensure / prove the functionality. The issue prominently arose in the setting of the example provided in the original issue. 

The discussion in the ticket outlined the issue and this change was made to address the behavior that was described in the ticket :)

Reply:
Got it. My concern is that if the tests pass even with the fix removed, then they aren't preventing us from regressing again.

I'm doing some refactoring of this code as part of https://github.com/CanCanCommunity/cancancan/pull/653 and I don't want to accidentally break it.

Do you think there is a test missing that I can include in that PR to ensure I don't regress?